Wikipedia/Images/WMFImageController.swift

```diff
@@ -124,14 +123,54 @@ public class WMFImageController : NSObject {
         .then(mainImageBlock)
     }
 
-    /// MARK: - Simple Fetching
+    /**
+    Use the shared session to download an image directly to the receiver's disk cache.
+
+    - parameter url: The URL to download the image from.
+
+    - returns: A promise which is resolved after the data has been successfully written to disk.
+    */
+    public func downloadImageDataInBackground(url: NSURL) -> Promise<Void> {
```

Comment:
I could investigate how much work it would be to add the capability of background, direct-to-disk downloads in `SDWebImage`, but my gut tells me it won't be easy and it doesn't get us extra affordances of using a background URL session (which has a lot more functionality than a simple `NSURLConnection` wrapped in a background task).